Linked packages cannot be found in 3.0

After updating our packages to 3.0, but not switching to federated build, we are able to successfully build and install our extensions but on `jupyter lab build` we get the following error for each use of a linked library by an extension.

```
ERROR in resolving fallback for shared module @elyra/application
Module not found: Error: Can't resolve 'file:linked_packages/elyra-application-2.0.0-dev-505867c7eec03453457e70c8d6b636759257fdd4.tgz' in '/Users/ajbozart/anaconda3/envs/elyra/share/jupyter/lab/staging/node_modules/@elyra/code-snippet-extension/lib'
```

We see this when running `make install` on the following branch of the Elyra project with JupyterLab rc5 installed: https://github.com/ajbozarth/elyra/tree/jupyterlab3-fork

--- a/dev_mode/webpack.config.js
+++ b/dev_mode/webpack.config.js
@@ -181,6 +181,20 @@ function ignored(path) {
   return ignore;
 }
 
+const shared = {
+  ...package_data.resolutions,
+  ...singletons
+}
+
+// Webpack module sharing expects version numbers, so if a resolution was a
+// filename, extract the right version number from what was installed
+Object.keys(shared).forEach(k => {
+  const v = shared[k];
+  if (typeof v === 'string' && v.startsWith('file:')) {
+    shared[k] = require(`${k}/package.json`).version;
+  }
+})
+
 const plugins = [
   new WPPlugin.NowatchDuplicatePackageCheckerPlugin({
     verbose: true,
@@ -206,10 +220,7 @@ const plugins = [
       name: ['_JUPYTERLAB', 'CORE_LIBRARY_FEDERATION']
     },
     name: 'CORE_FEDERATION',
-    shared: {
-      ...package_data.resolutions,
-      ...singletons
-    }
+    shared
   })
 ];
 

